django/utils/numberformat.py

```diff
@@ -26,6 +26,8 @@ def format(number, decimal_sep, decimal_pos=None, grouping=0, thousand_sep='',
         return mark_safe(number)
     # sign
     sign = ''
+    if isinstance(number, float) and 'e' in str(number).lower():
```

Comment:
I would just comment this. 

```
    # Treat potentially very large/small floats as Decimals. 
```

Reply:
Added.

Reply:
Out of curiosity, when does `'E'` is present in `str(number)`? Trying to figure out when the `lower()` call is necessary.

Reply:
@charettes Hmm. Maybe it doesn't... 🤔 My mum would call that "belt and braces".